src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h

```diff
@@ -129,38 +129,41 @@ class GenericThreadStackManagerImpl_OpenThread
     struct SrpClient
     {
         static constexpr uint8_t kMaxServicesNumber      = CHIP_DEVICE_CONFIG_THREAD_SRP_MAX_SERVICES;
-        static constexpr uint8_t kMaxInstanceNameSize    = chip::Mdns::kMdnsInstanceNameMaxSize;
-        static constexpr uint8_t kMaxNameSize            = chip::Mdns::kMdnsTypeAndProtocolMaxSize;
         static constexpr uint8_t kMaxHostNameSize        = 16;
         static constexpr const char * kDefaultDomainName = "default.service.arpa";
         static constexpr uint8_t kDefaultDomainNameSize  = 20;
         static constexpr uint8_t kMaxDomainNameSize      = 32;
 
 #if CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY
         // Thread supports both operational and commissionable discovery, so buffers sizes must be worst case.
-        static constexpr uint8_t kMaxTxtEntriesNumber = chip::Mdns::CommissionAdvertisingParameters::kNumAdvertisingTxtEntries;
-        static constexpr uint8_t kMaxTxtValueSize     = chip::Mdns::CommissionAdvertisingParameters::kTxtMaxValueSize;
-        static constexpr uint8_t kMaxTxtKeySize       = chip::Mdns::OperationalAdvertisingParameters::kTxtMaxKeySize;
+        static constexpr size_t kSubTypeMaxNumber    = Mdns::kSubTypeMaxNumber;
+        static constexpr size_t kSubTypeTotalLength  = Mdns::kSubTypeTotalLength;
+        static constexpr size_t kTxtMaxNumber        = Mdns::CommissionAdvertisingParameters::kTxtMaxNumber;
+        static constexpr size_t kTxtTotalValueLength = Mdns::CommissionAdvertisingParameters::kTxtTotalValueSize;
```

Comment:
If these are supposed to be worst-case, shouldn't they be std::max of the commission and operational parameters? This seems to be assuming the commissioning ones will be bigger, but that seems pretty fragile.

Reply:
Done